app/controllers/alchemy/admin/base_controller.rb

```diff
@@ -167,6 +167,21 @@ def is_page_preview?
       def load_locked_pages
         @locked_pages = Page.from_current_site.locked_by(current_alchemy_user)
       end
+
+      # Returns the current site.
+      #
+      def current_alchemy_site
+        @current_alchemy_site ||= begin
+          site_id = params[:site_id] || session[:site_id]
+          if site_id.nil?
+            session.delete :site_id
+            Site.find_for_host(request.host)
```

Comment:
I'm not sure that I understood right...
We load the site only if we don't have already the `@current_alchemy_site`